src/plugins/controls/duration/duration.js

```diff
@@ -70,66 +77,76 @@ export default pluginFactory({
                 return durationStore.getItem(attemptId);
             }
 
-            //one stopwatch to count the time
-            self.stopwatch = timerFactory({
-                autoStart: false
-            });
-
-            //update the duration on a regular basis
-            self.polling = pollingFactory({
-                action: function updateDuration() {
-                    //how many time elapsed from the last tick ?
+            /**
+             * Updates the duration of a particular item
+             *
+             * @returns {Promise}
+             */
+            function updateDuration() {
+                //how many time elapsed from the last tick ?
 
-                    var context = testRunner.getTestContext();
+                var context = testRunner.getTestContext();
 
-                    //store by attempt
-                    var itemAttemptId = `${context.itemIdentifier}#${context.attempt}`;
+                //store by attempt
+                var itemAttemptId = `${context.itemIdentifier}#${context.attempt}`;
 
-                    durationStore.getItem(itemAttemptId).then(function(duration) {
+                currentUpdatePromise = queue.serie(() => durationStore.getItem(itemAttemptId)
+                    .then(function(duration) {
                         var elapsed = self.stopwatch.tick();
                         duration = _.isNumber(duration) ? duration : 0;
                         elapsed = _.isNumber(elapsed) && elapsed > 0 ? elapsed / 1000 : 0;
 
                         //store the last duration
-                        durationStore.setItem(itemAttemptId, duration + elapsed);
-                    });
-                },
+                        return durationStore.setItem(itemAttemptId, duration + elapsed);
+                    })
+                );
+
+                return currentUpdatePromise;
+            }
+
+            //one stopwatch to count the time
+            self.stopwatch = timerFactory({
+                autoStart: false,
+            });
+
+            //update the duration on a regular basis
+            self.polling = pollingFactory({
+                action: updateDuration,
                 interval: refresh,
                 autoStart: false
             });
 
             //change plugin state
             testRunner
-
                 .after('renderitem', function() {
                     self.enable();
                 })
                 .on('enableitem', function() {
                     self.enable();
                 })
-
+                .before('move skip exit timeout error disableitem', function() {
+                    updateDuration().then(() => self.disable());
```

Comment:
could be good to catch it (if you don't want to report the error, at least log it)